api/desecapi/renderers.py

```diff
@@ -11,11 +12,23 @@ def render(self, data, media_type=None, renderer_context=None):
         response = renderer_context.get('response')
 
         if response and response.exception:
-            # TODO enable the renderer to handle responses where reponse.data is a list
-            if not isinstance(data, dict) or data.get('detail', None) is None:
-                raise ValueError('Expected response.data to be a dict with error details in response.data[\'detail\'], '
-                                 'but got %s:\n\n%s' % (type(response.data), response.data))
             response['Content-Type'] = 'text/plain'
-            return data['detail']
+
+            try:
+                return data['detail']
+            except KeyError:
+                pass
+
+            try:
+                details = [el['detail'] for el in data if getattr(el, 'detail', None) is not None]
+                if details:
+                    return ', '.join(details)
+            except TypeError:
```

Comment:
If we assume the structure of the elements in the list case to be the same as the structure of the single case, then we should also be checking for `KeyError` here (because we do so above)